src/vm/eventpipe.cpp

```diff
@@ -374,9 +374,19 @@ void EventPipe::DisableInternal(EventPipeSessionID id, EventPipeProviderCallback
     {
         // Disable the profiler.
         SampleProfiler::Disable();
+        
+        // Get the managed command line.
+        LPCWSTR pCmdLine = GetManagedCommandLine();
+
+        // If this process started as a service we won't have a managed command line
```

Comment:
What is the connection between starting as a service and not getting this command line? From what I can tell this comes from the arguments to CorHost2::ExecuteAssembly() and usage of the API doesn't seem correlated to being a service?

Reply:
The code I proposed is adopted from [`ProfileDataAllocateScenarioInfo`](https://github.com/dotnet/coreclr/blob/2e32f3b92f3a17b5821673babaa6883deb5ab209/src/vm/ceeload.cpp#L11771), I am not sure what "services" mean in that particular line, I just copied it.

In the test case, the function `CorHost2::ExecuteAssembly()` is never called in the second `corerun` process.

Reply:
Interesting, I'm not sure what it means there either : ) It sounds like the key piece is that hosting APIs can start the runtime in a mode that doesn't have a clear startup assembly and thus there is no fabricated managed command line.

The ProfileDataAllocateScenarioInfo looks to be part of the IBC training and it is generating a log that records what training scenario was run to generate specific data. Probably the string only needs to be human recognizable for display in a UI or log file in that case.